From 91d6000eef32ed4f1c895126a33e1c7d7483a25e Mon Sep 17 00:00:00 2001
From: ShootingStarDragons <ShootingStarDragons@protonmail.com>
Date: Wed, 10 May 2023 18:32:18 +0800
Subject: [PATCH 2/2] feat: damage text_input popup before commit

when move xdg, the text_input popup is updated, but itself is not
damaged yet, so before commit, damage it
---
 include/sway/input/text_input.h | 2 ++
 sway/desktop/output.c           | 3 +++
 sway/input/text_input.c         | 6 +++---
 3 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/include/sway/input/text_input.h b/include/sway/input/text_input.h
index 4320b8a5..2dbd5920 100644
--- a/include/sway/input/text_input.h
+++ b/include/sway/input/text_input.h
@@ -87,4 +87,6 @@ struct sway_text_input *sway_text_input_create(
 bool sway_input_popup_get_position(
         struct sway_input_popup *popup, int *lx, int *ly);
 
+void sway_input_popup_damage(struct sway_input_popup *popup);
+
 #endif
diff --git a/sway/desktop/output.c b/sway/desktop/output.c
index 5df4ba8f..6a29e3a1 100644
--- a/sway/desktop/output.c
+++ b/sway/desktop/output.c
@@ -271,6 +271,9 @@ void output_input_popups_for_each_surface(struct sway_output *output,
 			continue;
 		}
 
+		// damage the popup if surface is updated
+		sway_input_popup_damage(popup);
+
 		double ox = lx - output->lx;
 		double oy = ly - output->ly;
 
diff --git a/sway/input/text_input.c b/sway/input/text_input.c
index 2195c430..1d0e4109 100644
--- a/sway/input/text_input.c
+++ b/sway/input/text_input.c
@@ -33,7 +33,7 @@ static struct sway_text_input *relay_get_focused_text_input(
 }
 
 // damage the popup
-static void input_popup_damage(struct sway_input_popup *popup) {
+void sway_input_popup_damage(struct sway_input_popup *popup) {
 	if (!popup->visible) {
 		return;
 	}
@@ -68,7 +68,7 @@ static void input_popup_damage(struct sway_input_popup *popup) {
 }
 
 static void input_popup_update(struct sway_input_popup *popup) {
-	input_popup_damage(popup);
+	sway_input_popup_damage(popup);
 
 	struct sway_text_input *text_input =
 		relay_get_focused_text_input(popup->relay);
@@ -163,7 +163,7 @@ static void input_popup_update(struct sway_input_popup *popup) {
 			popup->popup_surface, &box);
 	}
 
-	input_popup_damage(popup);
+	sway_input_popup_damage(popup);
 }
 
 static void surface_send_enter_iterator(struct wlr_surface *surface,
-- 
2.39.3

